### PR TITLE
added delete rpc and docker-worker support

### PR DESCRIPTION
**Pull request by andreiSaw**

### CODE REVIEW

File: disneylandClient/worker.py [hunk 1]
```diff
@@ -93,6 +92,12 @@ def run(self):
...
-                p = Process(name=process_name, target=self.do_job, args=(job,))
+                p = Process(name=process_name, target=self.do_job, args=(job, queue))
                 self.processes.append(p)
                 p.start()
+
+            for p in self.processes:
+                p.join()
+                job = Job()
+                Parse(queue.get(), message=job)
+                self.stub.ModifyJob(job)
```
Comment (sashabaranov) [inline disneylandClient/worker.py]: This may raise exception and interrupt run

File: disneylandClient/worker.py [hunk 2]
```diff
@@ -93,6 +92,12 @@ def run(self):
 
                 self.acquire_cpus(process_name, 1)
 
-                p = Process(name=process_name, target=self.do_job, args=(job,))
+                p = Process(name=process_name, target=self.do_job, args=(job, queue))
                 self.processes.append(p)
                 p.start()
+
+            for p in self.processes:
+                p.join()
```
Comment (sashabaranov) [inline disneylandClient/worker.py]: This line will block for until child process terminates. Lets say you take first `p` in `self.processes` and wait for it. All other processes might already be finished, but you will block on this line. Moreover, this block will prevent worker from pulling new jobs.

**Review by sashabaranov — CHANGES_REQUESTED** [review]

**Comment by andreiSaw** [comment]
@sashabaranov updated

File: disneylandClient/disneyland_pb2_grpc.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Generated by the gRPC Python protocol compiler plugin. DO NOT EDIT!
 import grpc
 
-import disneylandClient.disneyland_pb2 as disneyland__pb2
+import disneyland_pb2 as disneyland__pb2
```
Comment (sashabaranov) [inline disneylandClient/disneyland_pb2_grpc.py]: this import does not work in py3

File: disneylandClient/worker.py [hunk 4]
```diff
@@ -93,6 +91,15 @@ def run(self):
 
                 self.acquire_cpus(process_name, 1)
 
-                p = Process(name=process_name, target=self.do_job, args=(job,))
+                p = Process(name=process_name, target=self.do_job, args=(job, self.queue))
```
Comment (sashabaranov) [inline disneylandClient/worker.py]: I recommend to just pass stub to worker func. It might want to modify job multiple times inside.

**Review by sashabaranov — CHANGES_REQUESTED** [review]

File: disneylandClient/worker.py (hunk 4 above)
Comment (andreiSaw) [reply disneylandClient/worker.py]: grpc not able to establish secure call from outside
screen shot 2017-11-02 at 16 59 25

File: disneylandClient/worker.py [hunk 5]
```diff
@@ -17,17 +16,25 @@ def __init__(
         self.job_kind = job_kind
         self.sleep_time = sleep_time
         self.do_job = job_func
-
         self.cpu_avail = threads_num
         self.cpus_per_job = {}  # job_id -> needed_cpus
         self.processes = []
-
+        self.queue = Queue()
```
Comment (sashabaranov) [inline disneylandClient/worker.py]: do we even need this now?

**Comment by andreiSaw** [comment]
@sashabaranov resolved

**Pull request merged**